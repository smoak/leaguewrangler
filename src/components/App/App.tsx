import CssBaseline from '@material-ui/core/CssBaseline';
import React, { SFC } from 'react';

import AppBody from '../AppBody';
import AppHeader from '../AppHeader';
import Login from '../Login';

export interface MappedStateProps {
  readonly isAuthenticated: boolean;
}

export type AppProps = MappedStateProps;

const App: SFC<AppProps> = ({ isAuthenticated }) => {
  return (
    <div>
      <CssBaseline />
      <AppHeader />
      {isAuthenticated ? <AppBody /> : <Login />}
    </div>
  );
};

export default App;
