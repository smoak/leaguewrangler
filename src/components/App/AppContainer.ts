import { MapStateToProps, connect } from 'react-redux';

import { isAuthenticated } from '../../data/domains/session/selectors';
import { ApplicationState } from '../../data/state/types';

import App, { MappedStateProps } from './App';

export const mapStateToProps: MapStateToProps<MappedStateProps, {}, ApplicationState> = state => {
  return {
    isAuthenticated: isAuthenticated(state),
  };
};

export default connect(mapStateToProps)(App);
