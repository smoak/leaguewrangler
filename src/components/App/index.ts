export { default } from './AppContainer';
export * from './App';
