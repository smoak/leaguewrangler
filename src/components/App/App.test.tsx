import { ShallowWrapper, shallow } from 'enzyme';

import App from './App';

describe('App', () => {
  let component: ShallowWrapper;

  describe('when not authenticated', () => {
    beforeEach(() => {
      component = shallow(<App isAuthenticated={false} />);
    });

    it('renders', () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe('when authenticated', () => {
    beforeEach(() => {
      component = shallow(<App isAuthenticated={true} />);
    });

    it('renders', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
