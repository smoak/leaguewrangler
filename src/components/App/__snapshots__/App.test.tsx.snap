// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders 1`] = `
<div
  className="App"
>
  <WithStyles(CssBaseline) />
  <div
    className="App__header"
  >
    <WithStyles(AppHeader) />
  </div>
  <div
    className="App__body"
  >
    <AppBody />
  </div>
</div>
`;
